react/features/large-video/components/LargeVideo.web.js

```diff
@@ -126,6 +166,32 @@ class LargeVideo extends Component<Props> {
         );
     }
 
+    _updateLayout: () => void;
+
+    /**
+     * Refreshes the video layout to determine the dimensions of the stage view.
+     * If the filmstrip is toggled it adds CSS transition classes and removes them
+     * when the transition is done.
+     *
+     * @returns {void}
+     */
+    _updateLayout() {
+        const { _verticalFilmstripWidth, _resizableFilmstrip } = this.props;
+
+        if (_resizableFilmstrip && _verticalFilmstripWidth >= FILMSTRIP_BREAKPOINT) {
+            this._containerRef.current.classList.add('transition');
+            this._wrapperRef.current.classList.add('transition');
+            VideoLayout.refreshLayout();
```

Comment:
Can't we move this to some middleware or something? It will look cleaner since it is legacy code...

Reply:
Not really, because we need to make sure that the transition class gets applied before the recalculation or the transition won't work